spec/unit/provider/mount/linux_spec.rb

```diff
@@ -0,0 +1,91 @@
+require "spec_helper"
+require "ostruct"
+
+describe Chef::Provider::Mount::Linux do
+  before(:each) do
+    @node = Chef::Node.new
+    @events = Chef::EventDispatch::Dispatcher.new
+    @run_context = Chef::RunContext.new(@node, {}, @events)
+
+    @new_resource = Chef::Resource::Mount.new("/tmp/foo")
+    @new_resource.device      "/dev/sdz1"
+    @new_resource.device_type :device
+    @new_resource.fstype      "ext3"
+
+    @new_resource.supports remount: false
+
+    @provider = Chef::Provider::Mount::Linux.new(@new_resource, @run_context)
+
+    allow(::File).to receive(:exists?).with("/dev/sdz1").and_return true
+    allow(::File).to receive(:exists?).with("/tmp/foo").and_return true
+    allow(::File).to receive(:realpath).with("/dev/sdz1").and_return "/dev/sdz1"
+    allow(::File).to receive(:realpath).with("/tmp/foo").and_return "/tmp/foo"
+  end
+
+  context "to see if the volume is mounted" do
+
+    it "should set mounted true if the mount point is found in the mounts list" do
+      allow(@provider).to receive(:shell_out!).and_return(OpenStruct.new(stdout: "/tmp/foo /dev/sdz1 type ext3 (rw)\n"))
```

Comment:
```suggestion
      allow(@provider).to receive(:shell_out!).and_return(double(stdout: "/tmp/foo /dev/sdz1 type ext3 (rw)\n"))
```

Since an OpenStruct instance will respond to any method, this can cause problems in the future when the code changes in unexpected ways. Instead the tests should be explicit about what behavior is expected of the test objects.

Reply:
Agreed, I have updated the changes.